steward/src/somm_send.rs

```diff
@@ -87,14 +87,22 @@ pub async fn data_hash(
     let mut hasher = sha2::Sha256::new();
     if let Some(vote) = &allocation.clone().vote {
         let mut buf = BytesMut::new();
-        vote.encode(&mut buf).unwrap();
+        vote.encode(&mut buf)?;
         let vote_data = hex::encode(&buf);
         let msg = format!("{}:{}:{}", allocation.salt, vote_data, val_address);
         hasher.update(msg.as_bytes());
 
         return Ok(AllocationPrecommit {
             hash: hasher.finalize().to_vec(),
-            cellar_id: vote.cellar.clone().unwrap().id,
+            cellar_id: vote
+                .cellar
+                .clone()
+                .ok_or_else::<Error, _>(|| {
+                    ErrorKind::AllocationError
```

Comment:
Shouldn't the presence of a cellar be checked at the time a vote is instantiated, as it would be invalid without it (e.g. it shouldn't be an `Option`)? Also, we'd want to check that the value is there before cloning. This would probably be cleaner pulling that value out before assigning it in the `return` statement.

Reply:
It is an `Option<T>` because that's how prost generates the RebalanceVote struct field. Will move that code up before the assignment.